snuba_sdk/delete_query.py

```diff
@@ -0,0 +1,46 @@
+import json
+from dataclasses import dataclass
+from typing import Any, Dict, List, Union
+
+from snuba_sdk.query import BaseQuery
+
+
+@dataclass(frozen=True)
+class DeleteQuery(BaseQuery):
+    """
+    This represents a snuba delete query.
+    Inputs:
+        storage - the storage to delete from
+        columnConditions - a mapping from column-name to a list of column values
+            that defines the delete conditions. ex:
+            {
+                "id": [1, 2, 3]
+                "status": ["failed"]
+            }
+            represents
+            DELETE FROM ... WHERE id in (1,2,3) AND status='failed'
+    Deletes all rows in the given storage, that satisfy the conditions
+    defined in 'columnConditions'.
+    """
+
+    storage_name: str
+    column_conditions: Dict[str, List[Any]]
+
+    def validate(self) -> None:
```

Comment:
I made the decision that input validation should not be done by the sdk, this is why I made that decision:
1. Input validation needs to be done (and is) by the actual snuba endpoint `/deletes`. By adding validation here as well we not only are writing duplicate code, but we are also creating 2 separate things that we need to keep updated now as opposed to 1.
2. The SDK already enforces proper types (with python typing and mypy) as long as these are respected then the client will be able to successfully deliver the request to the endpoint and recieve the proper response.

Reply:
I'm not sure if I agree that we shouldn't do validation here. I don't think it makes sense to send a query to snuba with something like `{"project_id": []}` for the columns, or to have columns be empty (`{}`) like you denote in the tests. 

It's true that we do validation in the endpoint as well, but it would be better to prevent making the request to snuba at all if we already know it's invalid. As for needing to keep these both updated, I would think that the two cases I raised above are not likely to change even if we add more validation on the endpoint.

Reply:
done